src/components/hub/elements/MessageInput.js

```diff
@@ -42,6 +43,36 @@ export default function MessageInput({
     [onSubmitFunc]
   );
 
+  const uploadInputText = useCallback(() => {
+    if (inputTextRef.current.value) {
+      localStorage.setItem("mtext", inputTextRef.current.value);
+      inputTextRef.current.value = "";
+      inputTextRef.current.style.height = `55px`;
+    }
+  }, [inputTextRef]);
+
+  const syncInputText = useCallback(() => {
+    const mtext = localStorage.getItem("mtext");
+    if (mtext) {
+      localStorage.removeItem("mtext");
+      inputTextRef.current.value = mtext;
+      handleInput({ target: { value: mtext } });
+    }
+  }, [inputTextRef, handleInput]);
+
+  useEffect(() => {
+    const isCurrentLocationAttach = location.hash.includes("/attach");
+
+    if (!isCurrentLocationAttach && isPrevLocationAttach) {
+      chatMessagesBlockRef.current?._infScroll?.scrollIntoView({
+        block: "end",
+      });
+      syncInputText();
+    }
+
+    setIsPrevLocationAttach(isCurrentLocationAttach);
+  }, [location, syncInputText, isPrevLocationAttach, chatMessagesBlockRef]);
```

Comment:
same here with chatMessagesBlockRef

Reply:
Also, so you have `isPrevLocationAttach ` as a dependency, and you also change it via `setIsPrevLocationAttach `. Is it gonna be an infinite loop?